lib/ecto/query/builder.ex

```diff
@@ -106,6 +106,10 @@ defmodule Ecto.Query.Builder do
     escape_with_type(expr, type, params_acc, vars, env)
   end
 
+  def escape({:type, _, [{:field, _, [_ | _]} = expr, type]}, _type, params_acc, vars, env) do
+    escape_with_type(expr, type, params_acc, vars, env)
+  end
```

Comment:
It feels like this is the only change you need. The way `validate_type!` worked with aliases should be fine. Or are you running into something?